packages/ibm-products/src/components/CoachmarkV2/CoachmarkBeacon/CoachmarkBeacon.tsx

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright IBM Corp. 2024, 2025
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+// Other standard imports.
+import PropTypes from 'prop-types';
+// Import portions of React that are needed.
+import React from 'react';
+import cx from 'classnames';
+import { getDevtoolsProps } from '../../../global/js/utils/devtools';
+import { pkg } from '../../../settings';
+import { ButtonProps, Button } from '@carbon/react';
+
+// The block part of our conventional BEM class names (blockClass__E--M).
+const blockClass = `${pkg.prefix}--coachmark-beacon`;
+const componentName = 'CoachmarkBeacon';
+
+const defaults = {
+  kind: 'default',
+};
+
+/**
+ * Reserved for future expansion, i.e. "RING".
+ */
+export enum BEACON_KIND {
+  DEFAULT = 'default',
+}
+
+export interface CoachmarkButtonProps extends ButtonProps<React.ElementType> {
+  onClick?(): void;
+  onDoubleClick?(): void;
+  tabIndex?: number;
+  ['aria-bool']?: boolean;
+}
+export interface CoachmarkBeaconProps {
+  /**
+   * Optional class name for this component.
+   */
+  className?: string;
+  /**
+   * What style of beacon.
+   * BEACON_KIND is an enum from the Coachmark and can be used for this value.
+   * @see {@link BEACON_KIND}
+   */
+  kind?: BEACON_KIND;
+  /**
+   * The aria label.
+   */
+  label: string;
+  /**
+   * button props
+   */
+  buttonProps?: CoachmarkButtonProps;
+}
+
+/**
+ * Use beacon for the target prop of a Coachmark component.
+ */
+export const CoachmarkBeacon = React.forwardRef<
+  HTMLDivElement,
+  CoachmarkBeaconProps
+>((props, ref) => {
+  const {
+    label,
+    className,
+    kind = defaults.kind,
```

Comment:
the `defaults` object isn't necessary. we can just define the default values here